### ISSUE TITLE
msys-mingw32

### ISSUE THREAD

**Issue by Anime4000** [frekky/iodine#18]
I using [XhmikosR's Builds](http://xhmikosr.1f0.de/tools/msys/) for MSYS (GCC 7.1.0)
and it has a bit older autoconf, I can simply update by get latest source and make install

however, for long time, your build a bit different, which is need `autoconf`

I run this command
```
autoreconf --install
export CC=x86_64-w64-mingw32-gcc
./configure
```
and this error came in
![windows 7 pro 64-bit-2017-07-02-08-00-41](https://user-images.githubusercontent.com/1908715/27766256-11e4a574-5efd-11e7-8afc-70e64481cee4.png)

sorry, I new into autoconf 🙇

**Comment by frekky** [comment]
Sorry, I can't really help you since I don't have a Windows installation available to test with! 
Since it looks like the `check` library is missing or `pkg-config` is not available (in the mingw build that you are using), it is probably easiest to just remove the following lines from `configure.ac` and rerun the commands you are using to build.
```
# The check library is used for the unit testing.  Not bothering to
# implement an --enable-tests option, as they are only run by "make
# check" so there is no user-level difference.  If the unit tests
# won't even *compile* there is probably a better solution than just
# turning them off, and build time is minimal.

PKG_CHECK_MODULES(CHECK,[check],
  AC_MSG_RESULT([enabled: unit testing]),
  AC_MSG_RESULT([disabled: unit testing]))

AM_CONDITIONAL([HAVE_CHECK],[test -z "${CHECK_PKG_ERRORS}"])
```
Let me know if that helps.

**Comment by Anime4000** [comment]
@frekky maybe you can install Windows via VM like VirtualBox.
However, after install `check` via https://libcheck.github.io/check/web/install.html#mingwsource
can't compile

**Comment by frekky** [comment]
What version of autotools are you using? If you can download and install a later version of autotools in mingw by source following a similar method, that might help. Also make sure you have `pkg-config` available. Did you have any luck by removing those lines from `configure.ac`?

**Comment by Anime4000** [comment]
I use [XhmikosR's Builds](http://xhmikosr.1f0.de/tools/msys/) for MSYS/Mingw
it has `pkg-config`
removing few lines `configure.ac`
can't compile, `check.h` missing, I sure install correctly

**Comment by Anime4000** [comment]
can you do [CMAKE](https://cmake.org/) for faster cross-compile, like [x265](https://bitbucket.org/multicoreware/x265) did

**Comment by frekky** [comment]
If you need to run iodine reliably and haven't got a working Windows build system, the easiest solution right now is to get a minimal Linux installation (ie. Debian 8) running on a VM and build/run inside the VM.

**Comment by Anime4000** [comment]
well, other option is using Raspberry Pi, will this iodine compile on ARM CPU?

**Comment by frekky** [comment]
Yes, it should do. If using Raspbian (or most Debian-based Linux distributions) you would need to have a number of packages installed first. The following should work, assuming you have an internet connection:
```
apt-get update
apt-get install build-essential check automake autoconf pkg-config
```
Then following the build instructions should result in a working binary.

**Comment by frekky** [comment]
Yes, it should do. Make sure you have the `build-essential` and `check` packages installed (if using Raspbian/Debian).

**Comment by frekky** [comment]
Yes, it should do. Make sure you have the `build-essential` and `check` packages installed (if using Raspbian/Debian).

**Comment by Anime4000** [comment]
Hello @frekky ... I successfully build your version of iodine under Windows, 

1. Using [msys2](http://www.msys2.org/)
2. Update msys2 by typing `pacman -Syu`, close terminal (no type exit)
3. Open `MSYS2 MSYS` at Start Menu and run `pacman -Su`
4. Install package `pacman -S base-devel git mingw-w64-x86_64-toolchain zlib-devel yasm cmake`

Terminal log here: https://gist.github.com/Anime4000/03bb3d18b0cfdfa9e2439e3dc045d0c2

I attach my version builds, can test on VM,
for my self, still not test yet, just having successful builds
[iodine-frekky.zip](https://github.com/frekky/iodine/files/1692088/iodine-frekky.zip)

**Comment by Anime4000** [comment]
I notice that you remove Windows code... no wonder it keep asking root even I run as admin 😄
